src/backend/impl/src/repositories/types/date_time.rs

```diff
@@ -19,6 +19,11 @@ impl DateTime {
         )?))
     }
 
+    pub fn from_timestamp_micros(micros: u64) -> Result<Self, ApiError> {
+        let dt = NaiveDateTime::from_timestamp_micros(micros.try_into().unwrap()).unwrap();
```

Comment:
Since this is used for logging, I think we need to be particularly careful with `unwrap`. It wouldn't be fun if the state was rolled back because log timestamps failed to be converted 😬 I don't think we need to handle this immediately though. We can come back and scan the codebase for `unwrap` or `expect` at any point.